### PR TITLE
circleci: Install packr2 and run before go build

### PR DESCRIPTION
**Pull request by zzak**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by zzak** [comment]
@BrewTestBot test this please

**Comment by zzak** [comment]
Pinging a couple people I've worked with before on these kinds of things just to hopefully get some movement on this, sorry for any extra noise :bow:

/cc @DomT4 @fxcoudert @commitay

File: Formula/circleci.rb [hunk 1]
```diff
@@ -19,10 +19,13 @@ def install
     ENV["GOPATH"] = buildpath
     ENV["GO111MODULE"] = "on"
 
+    system "go get -u github.com/gobuffalo/packr/v2/packr2"
```
Comment (fxcoudert) [inline Formula/circleci.rb]: This is not how we handle Go dependencies. Please look at other vending strategies in similar Go formulas.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by zzak** [comment]
@fxcoudert We made some changes to our build system to account for Homebrew's requirements. Would you mind taking another look? :bow:

**Comment by zzak** [comment]
@BrewTestBot test this please

**Comment by zzak** [comment]
@fxcoudert Would you mind reviewing this again please? :bow:

**Comment by fxcoudert** [comment]
`make dev` is downloading binaries behind our back:

```
==> make dev
bash .circleci/install-packr.sh
Fetching packr from https://github.com/gobuffalo/packr/releases/download/v2.0.1/packr_2.0.1_darwin_amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   621    0   621    0     0   1638      0 --:--:-- --:--:-- --:--:--  1642
100 4048k  100 4048k    0     0  1222k      0  0:00:03  0:00:03 --:--:-- 1713k
Installing packr for darwin_amd64 to /private/tmp/circleci-20190218-66662-1pnghji/src/github.com/CircleCI-Public/circleci-cli/bin
/private/tmp/circleci-20190218-66662-1pnghji/src/github.com/CircleCI-Public/circleci-cli/bin/packr2
```

**Comment by zzak** [comment]
Thanks for your reply @fxcoudert!

To give some more context here, we tried to update to the latest `circleci` in #36464 which failed because we started to use a new dependency [packr](https://github.com/gobuffalo/packr) for bundling static assets with our app. It's a common go dependency, so my first thought was to `go get -u` which was rejected here: https://github.com/Homebrew/homebrew-core/pull/36546#discussion_r254025390

As a workaround, we created a `make dev` task in CircleCI-Public/circleci-cli#272 that would install the correct version of `packr` based on the developers OS and platform to the latest stable version which we locked at `2.0.1`.

Alternatively, we could vendor their release binaries into git -- then perform a similar check for loading them during `make`.

Would that be acceptable for homebrew?

`packr` also has [their own tap](https://github.com/gobuffalo/homebrew-tap/blob/master/packr.rb), but I'm not sure how to integrate that with our formula here or if that is acceptable.

Thanks for your time and feedback!! :bow:

**Comment by fxcoudert** [comment]
Homebrew formulas build from source. If packr is widely used, then we'd love to have it as a formula in Homebrew (that builds from source). Then `circleci` can depend on it.

Basically anything that downloads binaries will be a no-go (except for cross-platform stuff, like Java and Mono), whether it's explicit or hidden in a makefile.

**Comment by zzak** [comment]
@fxcoudert Ok, thank you for your reply.

We understand this requirement for homebrew-core formulas and working on an alternative approach. We'll send a new PR when it's ready.

Thank you!! :bow:

**Pull request closed without merging**